### PR TITLE
JDK-8286338: suppress warnings about bad @author tags when author info is not generated.

### PR DESCRIPTION
**Pull request by jonathan-gibbons**
Please review a simple small change to `TagletManager`, to not report warnings about bad use of tags (such as `@author` and `@version`) when command-line options have not been given to enable the use of the tags in the output (i.e. the `-author` and the dreadfully-named `-version` option.)

The change is more pragmatic than anything else.  There are lots of "bad" uses of `@author` in JDK API documentation, on methods. The alternative would be to permit the tag anywhere, but that would be both a spec change and work to implement the change that we're not really interest in, since the tag is now somewhat deprecated.

Two existing tests are updated, to test the handling of "bad" tags, with and without the enabling option.

Another test has some very (very) old unused lines deleted. If today was Thursday, I'd call it Throwback Thursday. These lines predate the `JavadocTester` conversion.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8286338](https://bugs.openjdk.java.net/browse/JDK-8286338): suppress warnings about bad @author tags when author info is not generated.

### Reviewers
 * [Pavel Rappo](https://openjdk.java.net/census#prappo) (@pavelrappo - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8583/head:pull/8583` \
`$ git checkout pull/8583`

Update a local copy of the PR: \
`$ git checkout pull/8583` \
`$ git pull https://git.openjdk.java.net/jdk pull/8583/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8583`

View PR using the GUI difftool: \
`$ git pr show -t 8583`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8583.diff">https://git.openjdk.java.net/jdk/pull/8583.diff</a>

</details>

### CODE REVIEW

File: test/langtools/jdk/javadoc/doclet/testSimpleTagInherit/TestSimpleTagInherit.java [hunk 1]
```diff
@@ -35,16 +35,6 @@
 
 public class TestSimpleTagInherit extends JavadocTester {
 
-    //Javadoc arguments.
```
Comment (pavelrappo) [inline test/langtools/jdk/javadoc/doclet/testSimpleTagInherit/TestSimpleTagInherit.java:38]: This seems like an unrelated change; is it cleanup?

File: test/langtools/jdk/javadoc/doclet/testVersionTag/TestVersionTag.java [hunk 2]
```diff
@@ -53,7 +54,7 @@ public static void main(String... args) throws Exception {
         tb.writeJavaFiles(src,
                   """
                       package pkg;
-                      /** Introduction.\s
+                      /** Introduction.
```
Comment (pavelrappo) [inline test/langtools/jdk/javadoc/doclet/testVersionTag/TestVersionTag.java:57]: Cleanup?

**Review by pavelrappo — COMMENTED** [review]
What's that `-XDdummy=dummy` you use in tests?

File: test/langtools/jdk/javadoc/doclet/testVersionTag/TestVersionTag.java (hunk 2 above)
Comment (jonathan-gibbons) [reply test/langtools/jdk/javadoc/doclet/testVersionTag/TestVersionTag.java:57]: yes

**Comment by jonathan-gibbons** [comment]
> What's that `-XDdummy=dummy` you use in tests?

It's a no-op placeholder. JavaDoc does not allow empty arguments, and the alternative is dynamically build a list with a conditional component.  `-XD` is a "hidden" option to inject values in the compiler options map, so `-XDdummy=dummy` just injects a dummy value there.

File: test/langtools/jdk/javadoc/doclet/testSimpleTagInherit/TestSimpleTagInherit.java (hunk 1 above)
Comment (jonathan-gibbons) [reply test/langtools/jdk/javadoc/doclet/testSimpleTagInherit/TestSimpleTagInherit.java:38]: Yes, this is a holdover from the bad old days before `JavadocTester`, when tests were set up using fixed arrays like these. Frankly, I'm surprised these two empty arrays escaped notice for so long.

**Comment by pavelrappo** [comment]
> > What's that `-XDdummy=dummy` you use in tests?
> 
> It's a no-op placeholder. JavaDoc does not allow empty arguments, and the alternative is dynamically build a list with a conditional component. `-XD` is a "hidden" option to inject values in the compiler options map, so `-XDdummy=dummy` just injects a dummy value there.

Okay, I can see what happens when you pass an empty string:

    error: Illegal package name: ""

I wish there were a better way of conditionally providing a javadoc option.

**Review by pavelrappo — APPROVED** [review]
